### PR TITLE
OSD: move the warning to the cluster log when osd hitting the pg hard limit

### PR DESCRIPTION
**Pull request by taodd**
Fixes: https://tracker.ceph.com/issues/54050

Signed-off-by: dongdong tao <dongdong.tao@canonical.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by dvanders** [comment]
If we add something to the cluster log we should have a doc telling operator what to do

**Review by pdvian — COMMENTED** [review]
This does not seems to be vise option to log these messages to cluster log. Instead, we should raise it as a HEALTH_ERR (as client IOs are blocked) if PGs creation is withheld on OSD due to overdose protection of `mon_max_pg_per_osd * osd_max_pg_per_osd_hard_ratio`.

**Comment by vumrao** [comment]
cc @vumrao

File: src/osd/OSD.cc [hunk 1]
```diff
@@ -5051,8 +5051,8 @@ bool OSD::maybe_wait_for_max_pg(const OSDMapRef& osdmap,
     bool is_primary = osdmap->get_pg_acting_role(pgid, whoami) == 0;
     pending_creates_from_osd.emplace(pgid, is_primary);
   }
-  dout(1) << __func__ << " withhold creation of pg " << pgid
-	  << ": " << num_pgs << " >= "<< max_pgs_per_osd << dendl;
+  clog->warn() << "withhold creation of pg " << pgid
```
[the remark this answers is not in the document]
Comment (taodd) [reply src/osd/OSD.cc:5054]: Thanks @pdvian , that make sense, Raising a HEALTH_ERR would be better, I'll close this one!

**Comment by taodd** [comment]
Close this one, as we'll have https://github.com/ceph/ceph/pull/44962

**Pull request closed without merging**